### PR TITLE
LootProcess Timer

### PR DESCRIPTION
**Pull request by jandersson**
Return from `dispose_body` if a time delta is below threshold. Timer is reset upon looting

### CODE REVIEW

File: combat-trainer.lic [hunk 1]
```diff
@@ -481,8 +485,6 @@ class SpellProcess
     @osrel_no_harness = settings.osrel_no_harness
     echo("  @osrel_no_harness: #{settings.osrel_no_harness}") if $debug_mode_ct
 
-    @cast_only_to_train = settings.cast_only_to_train
```
Comment (jandersson) [inline combat-trainer.lic]: I'm a little confused about why these changes are still showing up. I've fetched and merged with the remote 'upstream' which is your master, and then created this branch

Comment (rcuhljr) [inline combat-trainer.lic]: Yeah, the timer commit removes these lines.

Comment (jandersson) [inline combat-trainer.lic]: Thanks, should be fixed now

File: combat-trainer.lic [hunk 2]
```diff
@@ -255,10 +260,10 @@ class LootProcess
       check_skinning(DRRoom.dead_npcs.first, game_state)
     end
     while 'and get ready to search it' == bput('loot', 'You search', 'I could not find what you were referring to', 'and get ready to search it')
+      @loot_timer = Time.now
```
Comment (rpherbig) [inline combat-trainer.lic]: I think this needs to be moved outside of the while loop. If you loot and get 'You search', the while exits without running the code inside, so `@loot_timer` never gets updated.

**Comment by jandersson** [comment]
Do some mobs decay at different rates than others?

File: combat-trainer.lic (hunk 2 above)
Comment (jandersson) [inline combat-trainer.lic]: The idea was to just update the timer if a looting is actually performed to act as a stopgap between the first check in `dispose_body` being `return if DRRoom.dead_npcs.empty?`

Comment (rpherbig) [inline combat-trainer.lic]: If a loot occurred successfully, the while loop exits and `@loot_timer` is never updated.

If loot occurs but fails, the while loop pauses and retries. In that case I do not think we would want to update `@loot_timer`.

Comment (jandersson) [inline combat-trainer.lic]: ah ha! OK rookie mistake

**Comment by rcuhljr** [comment]
Looks good, although we could probably safely use a lower loot timer, and at some point I wanted to just make this logic smarter and work off `DRRoom.dead_npcs.size`.

**Comment by rpherbig** [comment]
I ran this overnight and 5 seconds felt about right. I think we could lower it to 3-4 seconds without problem, but in practice you generally end up attacking or using a training ability between loots, which take long enough.
